src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php

```diff
@@ -162,9 +185,15 @@ public static function register($mode = 0)
                     return $b['count'] - $a['count'];
                 };
 
-                foreach (array('unsilenced', 'remaining', 'legacy', 'other') as $group) {
+                $groups = array('unsilenced', 'remaining');
+                if (DeprecationErrorHandler::MODE_WEAK_VENDORS === $mode) {
+                    $groups[] = 'remaining vendor';
+                }
+                array_push($groups, 'legacy', 'other');
+
+                foreach ($groups as $group) {
                     if ($deprecations[$group.'Count']) {
-                        echo "\n", $colorize(sprintf('%s deprecation notices (%d)', ucfirst($group), $deprecations[$group.'Count']), 'legacy' !== $group), "\n";
+                        echo "\n", $colorize(sprintf('%s deprecation notices (%d)', ucfirst($group), $deprecations[$group.'Count']), !in_array($group, array('legacy', 'remaining vendor'), true)), "\n";
```

Comment:
The value is: not repeating `$group`, and being ready for more groups. The downside is the `true` argument… will change as you say but again, matter of taste here IMO.